Fix for production view cache issue

fredricrylander opened the pull request:
I am currently staging a project for production that uses this plugin and ran into this bug.

It seems to only affect `fastify.view()` and not `reply.view()`. I stumbled upon it through the approach of supporting async/await mentioned here https://github.com/fastify/point-of-view/issues/87#issuecomment-465327052.

Regarding testing, I had to add an option to force the plugin into production mode. The tests were setup to fail and then clear as soon as the bug was fixed.

Eomm commented on index.js:
```diff
@@ -25,7 +25,7 @@ function fastifyView (fastify, opts, next) {
   const lru = HLRU(opts.maxCache || 100)
   const includeViewExtension = opts.includeViewExtension || false
-  const prod = process.env.NODE_ENV === 'production'
+  const prod = opts.prod || process.env.NODE_ENV === 'production'
```
We should add this option to the README

fredricrylander commented:
I described the option in README. Refactored so that it always overrides `NODE_ENV`, not only if set to `true`. I also renamed the option to `production` so that it goes more in line with other public options.

mcollina commented:
Ok for me.
